core/schema/container.graphqls

```diff
@@ -61,121 +80,439 @@ type Container {
+
+  Example:
+  withMountedFile(path: "/tmp/file.txt", source: "source-id")
+  """
+  withMountedFile(
+    """
+    Location of the mounted file (e.g., "/tmp/file.txt").
+    """
+    path: String!,
+
+    "Identifier of the mounted file."
+    source: FileID!
+  ): Container!
+
+  """
+  Retrieves this container plus a temporary directory mounted at the given path.
+
+  Example:
+  withMountedTemp(path: "/tmp/temp_dir")
+  """
+  withMountedTemp(
+    """
+    Location of the temporary directory (e.g., "/tmp/temp_dir").
+    """
     path: String!
-    cache: CacheID!
+  ): Container!
+
+  """
+  Retrieves this container plus a cache volume mounted at the given path.
+
+  Example:
+  withMountedCache(path: "/cache/node_modules", cache: "cache-id", source: "source-id")
+  """
+  withMountedCache(
+    """
+    Location of the cache directory (e.g., "/cache/node_modules").
+    """
+    path: String!,
+
+    "Identifier of the cache volume."
+    cache: CacheID!,
+
+    "Identifier of the directory to cache."
     source: DirectoryID
   ): Container!
 
-  "Retrieves this container plus a secret mounted into a file at the given path."
-  withMountedSecret(path: String!, source: SecretID!): Container!
+  """
+  Retrieves this container plus a secret mounted into a file at the given path.
 
-  "Retrieves this container after unmounting everything at the given path."
-  withoutMount(path: String!): Container!
+  Example:
+  withMountedSecret(path: "/tmp/secret.txt", source: "source-id")
+  """
+  withMountedSecret(
+    """
+    Location of the secret file (e.g., "/tmp/secret.txt").
+    """
+    path: String!,
 
-  "Retrieves this container plus the contents of the given file copied to the given path."
-  withFile(path: String!, source: FileID!, permissions: Int): Container!
+    "Identifier of the secret to mount."
+    source: SecretID!
+  ): Container!
 
-  "Retrieves this container plus a new file written at the given path."
-  withNewFile(path: String!, contents: String, permissions: Int): Container!
+  """
+  Retrieves this container after unmounting everything at the given path.
 
-  "Retrieves this container plus a directory written at the given path."
-  withDirectory(
+  Example:
+  withoutMount(path: "/cache/node_modules")
+  """
+  withoutMount(
+    """
+    Location of the cache directory (e.g., "/cache/node_modules").
+    """
     path: String!
-    directory: DirectoryID!
-    exclude: [String!]
+  ): Container!
+
+  """
+  Retrieves this container plus the contents of the given file copied to the given path.
+
+  Example:
+  withFile(path: "/tmp/file.txt", source: "source-id")
+  withFile(path: "/tmp/file.txt", source: "source-id", permissions: 0600)
+  """
+  withFile(
+    """
+    Location of the copied file (e.g., "/tmp/file.txt").
+    """
+    path: String!,
+
+    "Identifier of the file to copy."
+    source: FileID!,
+
+    """
+    Permission given to the copied file (e.g., 0600).
+
+    Default: 0644.
```

Comment:
Just a preference, not a blocker. I often see defaults in parenthesis after a longer description or as a [sentence after](https://docs.github.com/en/graphql/reference/queries#repository).

Example 1:

> Follow repository renames. If disabled, a repository referenced by its old name will return an error.
> **The default value is `true`.**

Example 2: 

> Follow repository renames. If disabled, a repository referenced by its old name will return an error.
> **Defaults to `true`.**

Example 3:

> Follow repository renames. If disabled, a repository referenced by its old name will return an error **(defaults to `true`)**.

By using a label it feels like there's special rendering for it, like in one language you could use `@default 0644`, in another it could be a markdown heading. Using prose is more neutral.

Again, it's just personal preference.

Reply:
Ok, I got it. In GraphQL docs, it doesn't have a standard except a sentence.

I guess we could separate from the rest of the body of the description and if the line contains default, we extract the default value from the sentence. But we need to have a standard way to phrase it.
I guess `Default to <value>.` should be easily parseable and be standard enough.